core/arch/mangle_shared.c

```diff
@@ -61,8 +61,8 @@ callee_info_t default_callee_info;
 int
 get_clean_call_switch_stack_size(void)
 {
-#ifdef AARCH64
-    /* Stack size needs to be 16 byte aligned on ARM */
+#if defined(AARCH64) || defined(X64)
+    /* Stack size needs to be 16 byte aligned on ARM and x64. */
```

Comment:
This is used only for out-of-line -- so yes this seems right to do for x64.  Inlined is aligned separately at the end of prepare_for_clean_call().  There the ifdef is x86_64 or MACOS -- no ARM, why not?

Also, please add || MACOS here to match the inlined.